app/views/topics/show.html.erb

```diff
@@ -82,22 +82,20 @@
               <% end %>
             </div>
           <% end %>
-          <% if @topic.child? %>
             <div class="govuk-!-margin-bottom-1">
-              <% if @topic.published? %>
+              <% if @topic.published? && !@topic.has_active_children?%>
```

Comment:
I wonder if this PR would benefit from implementing a `can_be_archived?` method on the `Tag` class.

This seems like a piece of business logic which the view 'knows', as well as the `TagArchiver` class, etc.

```suggestion
              <% if @topic.can_be_archived? %>
```

Reply:
Great idea! Simplifies the logic in `TagArchiver` and the views. We can add unit tests. 
Added in [ac01867](https://github.com/alphagov/collections-publisher/pull/1650/commits/ac01867b4ffa2ce54f2b5d0c08b7b52b41f3205c), with logic tweaks in the later commits.